src/plugins/intel_gpu/src/plugin/transformations/convert_precision_lrope.cpp

```diff
@@ -0,0 +1,85 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "convert_precision_lrope.hpp"
+
+#include "intel_gpu/op/gemm.hpp"
+
+#include "openvino/op/broadcast.hpp"
+#include "openvino/op/cos.hpp"
+#include "openvino/op/concat.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/convert.hpp"
+#include "openvino/op/reshape.hpp"
+#include "openvino/op/sin.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/pass/pattern/op/pattern.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "openvino/pass/pattern/op/or.hpp"
+#include "transformations/utils/utils.hpp"
+
+namespace ov::intel_gpu {
+
+ConvertPrecisionLRoPE::ConvertPrecisionLRoPE() {
```

Comment:
I think we can combine this with an existing logic src/plugins/intel_gpu/src/plugin/transformations/increase_position_ids_precision.cpp

Reply:
I merged this with the existing matcher(`IncreasePositionIdsPrecision`) as comment

Reply:
@andrew-k-park, did you have a chance to check why IncreasePositionIdsPrecision doesn't work for this model? From the description I thought your patch forces Rope calculation in FP32 data type, but it seems it just adds similar type upconversion for the subgraph between Broadcast and Sin/Cos nodes, so I'd expect it could be handled by IncreasePositionIdsPrecision. Can you please check what is the difference and share your model dumps?

Reply:
Removed unnecessary code additions by completely merging it into the existing MatcherPass by adding just relevant nodes in subgraph